### ISSUE TITLE
[loader] file path issue

### ISSUE THREAD

**Issue by kangfenmao** [llm-tools/embedJs#187]
### 🐛 Describe the bug

https://github.com/llm-tools/embedJs/blob/main/core/embedjs-utils/src/util/strings.ts

```ts
export function isValidURL(url: string) {
    try {
        new URL(url);
        return true;
    } catch {
        return false;
    }
}
```

### TestCase:

url C:\README.md
return true

### MarkdownLoader

this.isUrl = true

TypeError: fetch failed
[cause]: Error: unknown scheme

Here should be used file access mode to read the local directory files.

**Comment by adhityan** [comment]
Thank you for pointing the bug out. Updated signature to following in the upcoming release -

```ts
export function isValidURL(url: string) {
    try {
        const parsedUrl = new URL(url);
        return parsedUrl.protocol === 'http:' || parsedUrl.protocol === 'https:';
    } catch {
        return false;
    }
}
```

**Comment by adhityan** [comment]
Addressed in `0.1.26`
